src/watch/mod.rs

```diff
@@ -1,15 +1,252 @@
+            changes with the migration.\n"
+        );
+    }
+
+    print::msg!(
+        "{} {} for changes in:",
+        "Monitoring".emphasized(),
+        ctx.project.location.root.as_relative().display()
+    );
+    print::msg!("");
+    for m in &matchers {
+        print::msg!("  {}: {}", m.glob, m.target.to_string().muted());
+    }
+    print::msg!("");
+
+    let mut watcher = fs_watcher::FsWatcher::new()?;
+    // TODO: watch only directories that are needed, not the whole project
+
+    watcher.watch(&ctx.project.location.root, notify::RecursiveMode::Recursive)?;
+    let schema_dir = ctx.project.manifest.project().get_schema_dir();
+    if cmd.migrate && !schema_dir.starts_with(&ctx.project.location.root) {
+        watcher.watch(
+            &ctx.project.manifest.project().get_schema_dir(),
+            notify::RecursiveMode::Recursive,
+        )?;
+    }
+
+    loop {
+        let timeout = if let Some(retry_sec) = &cmd.retry_sec {
+            let any_failed = matchers.iter().any(|m| m.last_failed);
+            any_failed.then(|| tokio::time::Duration::from_secs(*retry_sec as u64))
+        } else {
+            None
+        };
+        if let Some(timeout) = timeout {
+            print::warn!("Retrying in {} sec...", timeout.as_secs());
+        }
+
+        if !cmd.continuous {
+            watcher.clear_queue();
+        }
+
+        // wait for changes
+        let event = watcher.wait(timeout).await;
+
+        let changed_paths = match event {
+            Event::Changed(paths) => paths,
+            Event::Retry => Default::default(),
+            Event::Abort => break,
+        };
+        // strip prefix
+        let changed_paths: Vec<_> = changed_paths
+            .iter()
+            .flat_map(|p| p.strip_prefix(&ctx.project.location.root).ok())
+            .map(|p| (p, globset::Candidate::new(p)))
+            .collect();
+
+        // run all matching scripts
+        for matcher in &mut matchers {
+            // does it match?
+            let (is_matched, reason) = if matcher.last_failed {
+                (true, "(retry)".to_string())
+            } else {
+                let matched_paths = changed_paths
+                    .iter()
+                    .filter(|x| matcher.matcher.is_match_candidate(&x.1))
+                    .map(|x| x.0.display().to_string())
+                    .collect::<Vec<_>>();
+                (!matched_paths.is_empty(), matched_paths.join(", "))
+            };
+
+            if !is_matched {
+                continue;
+            }
+
+            // print
+            print::msg!(
+                "{}",
+                format!(
+                    "--- {}: {} ---",
+                    matcher.glob,
+                    matcher.target.to_string().muted()
+                )
+            );
+            if cmd.verbose {
+                print::msg!("{}", format!("  triggered by: {reason}").muted());
```

Comment:
I don't have a strong feeling here, but I don't think we _need_ any special coloring for verbose output. Feel free to keep it though, just putting it out there.

Reply:
Maybe I should cut the files after 10th to: "a.log, b.log, +3 others" and drop the `.muted()`.

Reply:
Yeah, but I think making it muted doesn't make it any less text, or take up any less screen real-estate. But it certainly would help make the non-verbose text stand out more. Not sure if that's a desirable outcome, tbh, so I'm fine to leave it as-is. Just wanted to raise the question since you asked for feedback on colors, etc. FWIW, I'm probably more minimalist than most about colorful output.

Reply:
In my opinion, if `--verbose` isn't telling me everything I want to know, it's a waste. Maybe that means we should show this truncated version in the non-verbose mode and the full-output in verbose mode?

Reply:
FWIW, I'm also 100% fine with not printing the glob matches that triggered the change, too. Just seemed like a helpful bit of information since our watch mode is already outputting activity text.

Reply:
> if --verbose isn't telling me everything I want to know, it's a waste

That's true.

> Maybe that means we should show this truncated version in the non-verbose mode and the full-output in verbose mode?

I don't feel like matched files are really all that useful in a normal workflow. You know which files you've just edited. You are interested in what is now being triggered.

>  the glob matches that triggered the change

I am printing globs that have been triggered.